include/drivers/flash.h

```diff
@@ -79,6 +87,7 @@ __subsystem struct flash_driver_api {
 	flash_api_pages_layout page_layout;
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 	const size_t write_block_size;
+	flash_api_get_parameters get_parameters;
```

Comment:
OK, so since we're adding API unconditionally, this raises new questions.

`write_block_size` is clearly a parameter of the flash.  Shouldn't it move to this structure?

Then we can deprecate `flash_get_write_block_size()` and implement it temporarily using `flash_get_parameters()`.

I realize this starts down a slippery slope, but it looks like maybe we should start sliding.

Reply:
I have been thinking about moving the `write_block_size` to flash_parameters, though I am not sure whether this, as it would also men depreciation of existing API, in this PR.
@pabigot Should we do that now? Why is it slippery slope?

Reply:
Well, we don't *have* to deprecate the API, we can just implement in terms of the new API and leave it.  But it could be deprecated since it'd be redundant with the value in the parameter structure.

"Slippery" because we should also look at what's in the page layout description and see what else should be exposed.  Like whether there's a fixed minimum erase size.

I think that should be left for a separate PR, but that moving `write_block_size` is appropriate as it justifies the move to an aggregate parameter structure (we had just one value, now we have two, so we should generalize).

Reply:
OK, I will mote the `write_block_size` into the `flash_parameters` structure within this PR.